### ISSUE TITLE
[Feature request]: Partial auto-layout

### ISSUE THREAD

**Issue by airjp73** [statelyai/studio-issues#110]
Hi! I've been working with the editor for the first time today and overall it's pretty awesome to work with. Great work! :)

One minor pain point I've run into is that, as I'm building up my state machine, I like to move the nodes around into a layout that makes sense to me. But then as I add new nodes, I'm no longer able to leverage "Autolayout".

A more concrete example:
* Create "State A" node which has nested states & transitions.
* While working on "State A", Autolayout is pretty nice to have to keep things from getting cluttered
* Once "State A" is done, organize the layout within "State A" in a way that makes sense to me
* Now add "State B" which also has nested states & transitions

At this point, I would like to still be able to use Autolayout while I'm working on "State B", but if I do I'll lose the layout I made for "State A".

A few ideas for how this workflow could be:
* Drag a box around some nodes & choose "Autolayout selected nodes"
* Option to lock some nodes in their current position so Autolayout doesn't affect them
* Nested "Autolayout" option where you can autolayout only the nodes within a particular state node

**Comment by davidkpiano** [comment]
In Canny: https://statelyai.canny.io/editor/p/autolayout-improvements
